netty/src/main/java/io/grpc/transport/netty/Utils.java

```diff
@@ -91,10 +92,10 @@ static ByteBuf toByteBuf(ByteBufAllocator alloc, ByteBuffer source) {
   public static Metadata.Headers convertHeaders(Http2Headers http2Headers) {
     Metadata.Headers headers = new Metadata.Headers(convertHeadersToArray(http2Headers));
     if (http2Headers.authority() != null) {
-      headers.setAuthority(http2Headers.authority().toString());
+      headers.setAuthority(http2Headers.authority().toString(UTF_8));
```

Comment:
The `toString()` here isn't as dangerous as `String.toString()`. It is on Netty's `ByteString`. [It always uses Latin-1](https://github.com/netty/netty/blob/master/common/src/main/java/io/netty/util/ByteString.java#L696); not the default system `Charset`.

I'd suggest leaving it alone for now, since it isn't hurting much. Adding a comment to make it more clear could be a good idea though.